rust/connlib/tunnel/src/gateway.rs

```diff
@@ -351,6 +437,80 @@ impl GatewayState {
     }
 }
 
+fn handle_p2p_control_packet(
+    fz_p2p_control: FzP2pControlSlice,
+    peer: &ClientOnGateway,
+    buffered_events: &mut VecDeque<GatewayEvent>,
+) -> Option<IpPacket> {
+    use p2p_control::dns_resource_nat;
+
+    match fz_p2p_control.event_type() {
+        p2p_control::ASSIGNED_IPS_EVENT => {
+            let Ok(req) = dns_resource_nat::decode_assigned_ips(fz_p2p_control)
+                .inspect_err(|e| tracing::debug!("{e:#}"))
+            else {
+                return None;
+            };
+
+            if !peer.is_allowed(req.resource) {
+                tracing::debug!(cid = %peer.id(), resource = %req.resource, "Received `AssignedIpsEvent` for resource that is not allowed");
```

Comment:
This shouldn't happen during normal operation right? Client would wait until the client receive the ok response for the resource before sending this assigned ip event. So I think we should bump this log since it means something weird is going on?